src/modules/form/components/viewable/DynamicViewFields.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { ReactNode, useCallback } from 'react';
+
+import {
+  OverrideableDynamicFieldProps,
+  FormValues,
+  ItemMap,
+  LinkIdMap,
+  AdjustValueFn,
+} from '../../types';
+import { setDisabledLinkIdsBase } from '../DynamicFormFields';
+
+import DynamicViewField from './DynamicViewField';
+import DynamicViewGroup from './DynamicViewGroup';
+
+import {
+  DisabledDisplay,
+  FormDefinitionJson,
+  FormItem,
+  ItemType,
+  ServiceDetailType,
+  ValidationError,
+} from '@/types/gqlTypes';
+
+export interface Props {
+  definition: FormDefinitionJson;
+  errors?: ValidationError[];
+  warnings?: ValidationError[];
+  horizontal?: boolean;
+  bulk?: boolean;
+  visible?: boolean;
+  pickListRelationId?: string;
+  values: FormValues;
+  setValues: React.Dispatch<React.SetStateAction<FormValues>>;
+  itemMap: ItemMap;
+  autofillDependencyMap: LinkIdMap;
+  enabledDependencyMap: LinkIdMap;
+  disabledLinkIds: string[];
+  setDisabledLinkIds: React.Dispatch<React.SetStateAction<string[]>>;
+}
+
+export const isEnabled = (
+  item: FormItem,
+  disabledLinkIds: string[] = []
+): boolean => {
+  if (item.hidden) return false;
+  if (!item.enableWhen && item.item) {
+    // This is a group. Only show it if some children are enabled.
+    return item.item.some((i) => isEnabled(i, disabledLinkIds));
+  }
+  return !disabledLinkIds.includes(item.linkId);
+};
+
+export const isShown = (item: FormItem, disabledLinkIds: string[] = []) => {
+  if (
+    !isEnabled(item, disabledLinkIds) &&
+    item.disabledDisplay !== DisabledDisplay.Protected
+  )
+    return false;
+
+  return true;
+};
```

Comment:
These look like they are copied exactly from `DynamicFormFields`. Can you import them from there, or better yet move them to `formUtil.ts`